apps/labrinth/src/util/avalara1099.rs

```diff
@@ -0,0 +1,155 @@
+    pub attributes: T,
+    pub links: Option<HashMap<String, String>>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct FormResponse {
+    pub form_type: FormType,
+    pub form_id: Option<String>,
+    pub company_id: u32,
+    pub company_name: String,
+    pub company_email: String,
+    pub reference_id: String,
+    pub signed_at: Option<NaiveDateTime>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct W9FormsResponse {
+    pub e_delivery_consented_at: Option<NaiveDateTime>,
+    pub tin_match_status: Option<String>,
+    pub entry_status: String,
+}
+
+pub async fn request_form(
+    user_id: DBUserId,
+    form_type: FormType,
+) -> Result<Result<DataWrapper<FormResponse>, serde_json::Value>, ApiError> {
+    const DEFAULT_TTL: u32 = 3600;
+
+    #[derive(Serialize, Deserialize)]
+    struct FormRequest {
+        form_type: FormType,
+        company_id: String,
+        reference_id: String,
+        ttl: u32,
+    }
+
+    let (request_builder, company_id) =
+        team_request(reqwest::Method::POST, "/form_requests")?;
+
+    let response = request_builder
+        .json(&DataWrapper {
+            data: Data {
+                r#type: Some("form_request".to_owned()),
+                id: None,
+                attributes: FormRequest {
+                    form_type,
+                    company_id,
+                    ttl: DEFAULT_TTL,
+                    reference_id: Reference {
+                        user_id,
+                        form_type,
+                        current_year: chrono::Utc::now().year_ce().1,
+                    }
+                    .to_string(),
+                },
+                links: None,
+            },
+        })
+        .send()
+        .await?;
+
+    Ok(if response.status().is_success() {
+        Ok(response.json::<DataWrapper<FormResponse>>().await?)
+    } else {
+        Err(response.json().await?)
+    })
+}
+
+pub async fn check_form(
+    reference_id: &str,
+) -> Result<
+    Result<Option<DataWrapper<W9FormsResponse>>, serde_json::Value>,
+    ApiError,
+> {
+    let (request_builder, _company_id) = team_request(
+        reqwest::Method::GET,
+        &format!(
+            "/w9forms?filter[reference_id_eq]={reference_id}&page[number]=1&page[size]=1"
+        ),
+    )?;
+
+    let response = request_builder.send().await?;
+
+    Ok(if response.status().is_success() {
+        let mut list_wrapper =
+            response.json::<ListWrapper<W9FormsResponse>>().await?;
+
+        Ok(list_wrapper.data.pop().map(|data| DataWrapper { data }))
+    } else {
+        Err(response.json().await?)
+    })
+}
+
+fn team_request(
+    method: reqwest::Method,
+    route: &str,
+) -> Result<(reqwest::RequestBuilder, String), ApiError> {
+    let key = dotenvy::var("AVALARA_1099_API_KEY")?;
+    let url = dotenvy::var("AVALARA_1099_API_URL")?;
+    let team = dotenvy::var("AVALARA_1099_API_TEAM_ID")?;
+    let company = dotenvy::var("AVALARA_1099_COMPANY_ID")?;
```

Comment:
These environment variables should be added to the sample Labrinth `.env` files, and ensured they are validated for being defined in `apps/labrinth/src/lib.rs#check_env_vars`. It's a bit late to error out due to a bad deployment configuration only when the time to actually send an API request comes.